src/Integration/Binding/CSharpVBProjectBinder.cs

```diff
@@ -87,19 +97,27 @@ public bool IsBindingRequired(BindingConfiguration binding, Project project)
 
         private bool IsFullyBoundProject(BindingConfiguration binding, Project project, Language language)
         {
-            if (!IsSolutionBound(binding, language, out var solutionRuleSetFilePath, out var additionalFilePath))
+            logger.LogDebug($"[Binding check] Checking binding for project '{project.Name}', language '{language}'");
+
+            var isSolutionBound = IsSolutionBound(binding, language, out var solutionRuleSetFilePath, out var additionalFilePath);
+            logger.LogDebug($"[Binding check] Is solution bound: {isSolutionBound}");
+
+            if (!isSolutionBound)
             {
+                logger.LogDebug("[Binding check] Solution is not bound. Skipping project-level checks.");
```

Comment:
The C#/VB checks look at multiple conditions, so I've logged data about each check.